src/Symfony/Component/Console/Tests/Helper/ProcessHelperTest.php

```diff
@@ -23,17 +23,65 @@ class ProcessHelperTest extends TestCase
     /**
      * @dataProvider provideCommandsAndOutput
      */
-    public function testVariousProcessRuns(string $expected, Process|string|array $cmd, int $verbosity, ?string $error)
+    public function testVariousProcessRuns(array $expectedOutputLines, bool $successful, Process|string|array $cmd, int $verbosity, ?string $error)
     {
         if (\is_string($cmd)) {
             $cmd = Process::fromShellCommandline($cmd);
         }
 
         $helper = new ProcessHelper();
         $helper->setHelperSet(new HelperSet([new DebugFormatterHelper()]));
-        $output = $this->getOutputStream($verbosity);
-        $helper->run($output, $cmd, $error);
-        $this->assertEquals($expected, $this->getOutput($output));
+        $outputStream = $this->getOutputStream($verbosity);
+        $helper->run($outputStream, $cmd, $error);
+
+        $expectedLines = 1 + \count($expectedOutputLines);
+
+        if (StreamOutput::VERBOSITY_VERY_VERBOSE <= $verbosity) {
+            // the executed command and the result are displayed
+            $expectedLines += 2;
+        }
+
+        if (null !== $error) {
+            ++$expectedLines;
+        }
+
+        $output = explode("\n", $this->getOutput($outputStream));
+
+        $this->assertCount($expectedLines, $output);
+
+        // remove the trailing newline
+        array_pop($output);
+
+        if (null !== $error) {
+            $this->assertSame($error, array_pop($output));
+        }
+
+        if (StreamOutput::VERBOSITY_VERY_VERBOSE <= $verbosity) {
+            if ($cmd instanceof Process) {
+                $expectedCommandLine = $cmd->getCommandLine();
+            } elseif (\is_array($cmd) && $cmd[0] instanceof Process) {
+                $expectedCommandLine = $cmd[0]->getCommandLine();
+            } elseif (\is_array($cmd)) {
+                $expectedCommandLine = (new Process($cmd))->getCommandLine();
+            } else {
+                $expectedCommandLine = $cmd;
+            }
+
+            $this->assertSame('  RUN  '.$expectedCommandLine, array_shift($output));
+
+            if ($successful) {
+                $this->assertSame('  RES  Command ran successfully', array_pop($output));
+            } else {
+                $this->assertSame('  RES  252 Command did not run successfully', array_pop($output));
+            }
+        }
+
+        if ([] !== $expectedOutputLines) {
+            sort($expectedOutputLines);
+            sort($output);
```

Comment:
This is the actual fix for failures like https://github.com/symfony/symfony/actions/runs/17972860504/job/51119399135#step:8:1127. We do not care about whether lines on STDOUT or STDERR are captured first.